client/src/hooks/useInfiniteScroll.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect } from 'react';
+
+interface UseInfiniteScrollParams {
+  scrollTargetRef: React.RefObject<HTMLDivElement>;
+  debounceTreshhold?: number;
+  scrollTreshhold?: number;
+  handleScroll: () => void;
+}
+
+// eslint-disable-next-line
```

Comment:
Is it still nececery?

Reply:
Yes, as debounce should accept any array and pass them without constraining the type